### PR TITLE
Xmlrpc/MetaWeblog

### LINKED ISSUE #381: Support MetaWeblog API
To provide the ability to publish with live Writer, we need to implement the MetaWeblog API.

Will we use the same concept than in O1?

The XmlRpc module allows to generate the LiveWriter/Manifest and the rsd xml description of the apis supported.
In different modules (Body, Title, Tags, ...), in XmlRpcHandler.cs, the response is processed depending on the XmlRpc method called.

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes #381

### CODE REVIEW

File: src/Orchard.Cms.Web/Modules/Orchard.Lists/Module.txt [hunk 1]
```diff
@@ -5,7 +5,9 @@ Website: http://orchardproject.net
...
 Description: Introduces a preconfigured container-enabled content type.
-FeatureDescription: A basic container-enabled content type.
-Dependencies: Orchard.Contents, Orchard.ContentTypes
-Category: Content Management
-
+Features:
+    Orchard.Lists:
+        Name: Lists
+        Description: Introduces a preconfigured container-enabled content type.
+        Categories: Content Management
```
Comment (agriffard) [inline src/Orchard.Cms.Web/Modules/Orchard.Lists/Module.txt]: Categories => Category

File: src/Orchard.Cms.Web/Modules/Orchard.XmlRpc/Module.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Features:
+    Orchard.XmlRpc:
+        Name: XML-RPC
+        Description: The XML-RPC module enables some local features to be used remotely.
+        Categories: Infrastructure
+    Orchard.RemotePublishing:
+        Name: Remote Publishing
+        Description: The remote publishing feature enables creation of contents from client applications such as Open Live Writer.
+        Dependencies: Orchard.XmlRpc
+        Categories: Infrastructure
```
Comment (agriffard) [inline src/Orchard.Cms.Web/Modules/Orchard.XmlRpc/Module.txt]: Categories => Category or it will appear as Uncategorized in modules admin.

**Comment by agriffard** [comment]
When I 'post draft to blog' with OLW, it should create a draft but it doesn't seem to be done.

**Comment by sebastienros** [comment]
I haven't tried this scenario, will fix it. There might also be an issue with CreatedUtc ..

**Comment by agriffard** [comment]
The draft is created in database. I don't know why it is not displayed in the blog admin list.

**Comment by sebastienros** [comment]
Fixed, it was not related to metaweblog api

**Comment by sebastienros** [comment]
Also side note, the theme support in OLW only works if the summary view renders the body. TheBlogTheme doesn't by default. I filed a bug in OLW there is nothing we can do about it. To test I rendered the body from the summary.

**Comment by Jetski5822** [comment]
For the extension manager stuff, could you add a test for it. I understand the use case, just thought the implementation was slightly different

**Comment by sebastienros** [comment]
There are currently no tests for `OrchardFeatureAttribute`, and I am too lazy to create one for now. I assume it also requires separate assemblies, and we have to mock the `ExportedTypes` property.

**Pull request merged**